### PR TITLE
Issue 616: Update k8 and Controller-run time versions 

### LINKED ISSUE #616: Update k8 and Controller-run time versions
### Description

Currently the controller-run time version used in operator is 0.6.5 and K8 version is 1.19. Update these to latest versions. Also remove replace directives from go.mod

### Importance

must-have

### Location

go.mod
Also update affected modules

### Suggestions for an improvement

upgrade version of controller-runtime to 0.11.0

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anishakj <anisha.kj@dell.com>

### Change log description

Updating controller-runtime to `11.0` and k8 version to `23.0`
### Purpose of the change

 Fixes #616

### What the code does

Updated the versions in `go.mod` and removed replace directives
Changed the Api calls accordance with new controller run time
Rewrite the e2e tests using `envtest`

### How to verify it
All unit and e2e tests should pass

### CODE REVIEW

**Review by tmarcu — APPROVED** [review]
mostly suggestions

File: main.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	webhookFlag bool
+	log         = ctrl.Log.WithName("cmd")
+	scheme      = apimachineryruntime.NewScheme()
+)
+
+func init() {
+	flag.BoolVar(&versionFlag, "version", false, "Show version and quit")
+	flag.BoolVar(&controllerconfig.TestMode, "test", false, "Enable test mode. Do not use this flag in production")
+	flag.BoolVar(&webhookFlag, "webhook", true, "Enable webhook, the default is enabled.")
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
```
Comment (nishant-yt) [inline main.go:55]: Need to add schemes related to Pravegacluster type also
`utilruntime.Must(v1beta1.AddToScheme(scheme))`

**Review by nishant-yt — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (nishant-yt) [inline main.go:55]: Need to add schemes related to Pravegacluster type also
`utilruntime.Must(v1beta1.AddToScheme(scheme))`

**Review by eddiepavkovic — APPROVED** [review]
gg

File: main.go (hunk 1 above)
Comment (anishakj) [reply main.go:55]: done

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # http://www.apache.org/licenses/LICENSE-2.0
 
 SHELL=/bin/bash -o pipefail
-
+CRD_OPTIONS ?= "crd:trivialVersions=true"
```
[the remark this answers is not in the document]
Comment (anishakj) [reply Makefile:11]: done

File: controllers/upgrade_test.go [hunk 3]
```diff
@@ -8,17 +8,16 @@
...
 
 import (
 	"context"
 	"strings"
-	"testing"
 
-	"github.com/pravega/pravega-operator/pkg/apis/pravega/v1beta1"
-	"github.com/pravega/pravega-operator/pkg/controller/pravega"
+	"github.com/pravega/pravega-operator/api/v1beta1"
+	//"github.com/pravega/pravega-operator/pkg/controller/pravega"
```
[the remark this answers is not in the document]
Comment (anishakj) [reply controllers/upgrade_test.go:18]: done

File: test/e2e/external_access_basic_test.go [hunk 4]
```diff
@@ -11,54 +11,40 @@
...
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
 	pravega_e2eutil "github.com/pravega/pravega-operator/pkg/test/e2e/e2eutil"
 )
 
-// Test create and recreate a Pravega cluster with the same name
-func testExternalCreateRecreateCluster(t *testing.T) {
-	g := NewGomegaWithT(t)
+var _ = Describe("External Access tests", func() {
+	Context("Creating cluster with external access", func() {
+		It("Creation of pods should be successful", func() {
```
[the remark this answers is not in the document]
Comment (anishakj) [reply test/e2e/external_access_basic_test.go]: done

**Pull request merged**